File: LeetCode/01.two_sum.py
# https://leetcode.com/problems/two-sum/description/

class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        if len(nums) <= 1:
            return False
        map = {}
        for index in range(len(nums)):
            if nums[index] in map:
                return [map[nums[index]], index]
            else:
                map[target - nums[index]] = index

File: LeetCode/01.two_sum.swift
// https://leetcode.com/problems/two-sum/description/

class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        if (nums.count <= 1) { return [-1] }

        var dict = [Int: Int]()
        for (index, num) in nums.enumerated() {
            if let returnIndex = dict[num] {
                return [returnIndex, index]
            } else {
                dict[target - num] = index
            }
        }
        return [-1]
    }
}
